src/bootstrap/src/ferrocene/dist/flip_link.rs

```diff
@@ -0,0 +1,30 @@
+use crate::builder::{Builder, RunConfig, ShouldRun, Step};
+use crate::core::config::TargetSelection;
+use crate::utils::tarball::{GeneratedTarball, Tarball};
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub(crate) struct FlipLink {
+    pub(crate) target: TargetSelection,
+}
+
+impl Step for FlipLink {
+    type Output = GeneratedTarball;
+    const DEFAULT: bool = true;
+    const ONLY_HOSTS: bool = true;
+
+    fn should_run(run: ShouldRun<'_>) -> ShouldRun<'_> {
+        run.alias("flip-link")
+    }
+
+    fn make_run(run: RunConfig<'_>) {
+        run.builder.ensure(FlipLink { target: run.target });
+    }
+
+    fn run(self, builder: &Builder<'_>) -> Self::Output {
+        let flip_link = builder.ensure(crate::ferrocene::tool::FlipLink { target: self.target });
+
+        let tarball = Tarball::new(builder, "flip-link", &self.target.triple);
+        tarball.add_file(flip_link, "bin", 0o755);
```

Comment:
Should flip-link be something users invoke directly? If so, we should register it as a binary proxy.

Reply:
Not directly, but they would configure the compiler to invoke it. See https://github.com/knurling-rs/flip-link?tab=readme-ov-file#usage.

```toml
# .cargo/config.toml

[target.'cfg(all(target_arch = "arm", target_os = "none"))']
# (..)
linker = "flip-link"

# OR

[target.'cfg(all(target_arch = "arm", target_os = "none"))']
# (..)
rustflags = [
  "-C", "linker=flip-link",
  # (..)
]
```

Reply:
So should we add it? (not exactly sure what a binary proxy is tbh)

Reply:
Added it